2019

hishamnajam opened the pull request:
The dataset gives the following two warnings during its validation on the Gapminder Offline Tool:

> Entity value should not be equal to entity domain name or entity set name. This rule providing is critical for DDFQL and DDF reader supporting (on WS).

> GEO and landlocked

 
The dataset also doesn't open properly on the Gapminder Offline Tool. Probably its due to the above warning. 

I checked the data for each CSV individually on the Gapminder Offline Tool, and it all seems well.


- The indicators are updated to the latest 2019 data published by BP Statiscal Review

- Coal Consumption and Natural Gas Production indicators are now reported in exajoules. Previously they were reported in Million Tons of Oil Equivalent (mtoe).

hishamnajam committed:
Ran ddf cleanup

hishamnajam committed:
Running recipe - 1st attempt

hishamnajam committed:
Recipe work, but fails to find pop-datapoints

hishamnajam committed:
Successful Recipe etl.yaml run.

hishamnajam committed:
added ddf files to branch

hishamnajam committed:
Data verified, All units satisfied.

hishamnajam committed:
Updated the Readme

jheeffer commented on etl/recipes/etl.yaml:
```diff
@@ -75,14 +79,14 @@ cooking:
-                  oil_proved_reserves_total: "oil_proved_reserves_history * 136400000"
-                  natural_gas_proved_reserves_total: "gas_proved_reserves_history * 900000000"
+                  natural_gas_production_total: "gas_production_exajoules"
+                  gas_proved_reserves: "gas_proved_reserves * 900000000"
```
Is removing `_history` from this indicator on purpose?
Upon closer inspection it seems this `gas_proved_reserves` isn't used at all; it is filtered out in next procedure. Guess it can go?

hishamnajam replied:
> Is removing _history from this indicator on purpose?

Yes, it was already done in the etl script of _ddf--bp--energy_ , so I updated it here accordingly.
_gas_proved_reserves_history --> gas_proved_reserves_ 

I'll remove _gas_proved_reserves_ , and put this info into commit message for the record.

jheeffer commented:
> Entity value should not be equal to entity domain name or entity set name. This rule providing is critical for DDFQL and DDF reader supporting (on WS).

> GEO and landlocked

This was specifically because of a limitation/bug in the old database server (Waffle Server). It can be ignored for our current infrastructure (Big Waffle). We've never really built validator ourselves (was done by Valor) so changing this will be a time-intensive task (getting to know the codebase & build/publish processes).

> The dataset also doesn't open properly on the Gapminder Offline Tool. Probably its due to the above warning.

There seems to be a bug in Vizabi that makes it request tags even when the dataset doesn't contain any tags. This makes the reader crash. So it has nothing to do with the dataset validity.
I temporarily fixed it by adding a file `ddf--entities--tag.csv` with content `tag,name` and regenerate datapackage.json. That removes the error from Tools Offline. But we shouldn't have to add this file to every dataset, so no need to make that part of this (or any future) PR.

cc @angieskazka 🐛 

Apart from the comment on `gas_proved_reserves`, dataset looks good.

hishamnajam committed:
gas_proved_reserves_history --> gas_proved_reserves
It was already done in the etl script of ddf--bp--energy , so I updated it here accordingly.

Also,
gas_proved_reserves isn't used at all; it is filtered out in next procedure. So it can go!

hishamnajam committed:
Coal Consumption and Natural Gas Production indicators are now converted from Exajoules to Tons of Oil Equivalent (toe) (1 toe = 41.868 million kJ, BP). Previously they were reported in Million Tons of Oil Equivalent (mtoe) and converted to Tons of Oil Equivalent (toe).
Unit conversion is done in the etl.yaml .

jheeffer commented on datapackage.json:
```diff
@@ -935,6 +950,15 @@
             }
         ],
         "entities": [
+            {
```
Looks like you didn't update datapackage.json after removing `tag` entities ; ) I assume you tried it out for gapminder offline

hishamnajam replied:
Yup, and it worked perfectly, Thanks for the tip. I'll remove it, and update the datapackage.json.

hishamnajam committed:
Removed ddf--entities--tag.csv and updated the datapackage.json
